### DISCUSSION TITLE
Hi, Can you elaborate on the microstructure generation algorithm?

### DISCUSSION THREAD

**Discussion by OldWheat** [materialsinnovation/pymks#575] [General]
I want to know the microstructure generation algorithm of the IPA `make_microstructure`. Can you tell me how the microstructure generation is implemented and provide some relevant references

**Comment by beyucel** [comment]
Hi,
 The microstructure generation process starts by assigning a random number within the range of [0, 1] to every voxel in the MVE (2-D or 3-D).Then, a Gaussian filter is applied on this random number field while treating boundaries as periodic to produce periodic (discretized) fields (i.e., the values wrap around the boundaries without any marked discontinuties). The desired discretized MVEs are obtained by simply thresholding the filtered image to attain a preselected volume fraction.

Data generation section clearly explains the protocol (with figures).

https://www.osti.gov/pages/biblio/1559631

**Reply by OldWheat** [reply]
Thank you so much, this answer is very helpful!
